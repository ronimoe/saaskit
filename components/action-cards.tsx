import { Card, CardContent } from "@/components/ui/card"
import { Download, Upload, Star, Heart } from "lucide-react"

export function ActionCards() {
  const actions = [
    {
      icon: Download,
      title: "Download",
      description: "Export your data",
      color: "text-blue-600"
    },
    {
      icon: Upload,
      title: "Upload",
      description: "Import new files",
      color: "text-green-600"
    },
    {
      icon: Star,
      title: "Favorites",
      description: "Saved items",
      color: "text-yellow-600"
    },
    {
      icon: Heart,
      title: "Liked",
      description: "Your preferences",
      color: "text-red-600"
    }
  ]

  return (
    <section className="mb-12">
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4">
        {actions.map((action) => {
          const Icon = action.icon
          return (
            <Card key={action.title} className="cursor-pointer hover:shadow-lg transition-shadow">
              <CardContent className="p-6 text-center">
                <Icon className={`w-8 h-8 mx-auto mb-2 ${action.color}`} />
                <h3 className="font-semibold mb-1">{action.title}</h3>
                <p className="text-sm text-slate-600 dark:text-slate-300">{action.description}</p>
              </CardContent>
            </Card>
          )
        })}
      </div>
    </section>
  )
} 