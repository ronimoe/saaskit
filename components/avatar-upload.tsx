'use client'

import { useState, useRef, useCallback } from 'react'
import { Upload, Loader2, X, Camera, Check } from 'lucide-react'
import { Button } from '@/components/ui/button'
import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar'
import { Dialog, DialogContent, DialogHeader, DialogTitle, DialogTrigger } from '@/components/ui/dialog'
import { createClient } from '@/utils/supabase/client'
import { useNotifications } from '@/components/providers/notification-provider'
import { cn } from '@/lib/utils'

interface AvatarUploadProps {
  currentAvatarUrl?: string | null
  userDisplayName: string
  userId: string
  onAvatarUpdate?: (newAvatarUrl: string) => void
  className?: string
}

const MAX_FILE_SIZE = 5 * 1024 * 1024 // 5MB
const ALLOWED_FILE_TYPES = ['image/jpeg', 'image/jpg', 'image/png', 'image/webp']

function getInitials(name: string): string {
  return name
    .split(' ')
    .map(word => word.charAt(0))
    .join('')
    .toUpperCase()
    .slice(0, 2)
}

export function AvatarUpload({ 
  currentAvatarUrl, 
  userDisplayName, 
  userId, 
  onAvatarUpdate,
  className 
}: AvatarUploadProps) {
  const [isOpen, setIsOpen] = useState(false)
  const [selectedFile, setSelectedFile] = useState<File | null>(null)
  const [previewUrl, setPreviewUrl] = useState<string | null>(null)
  const [isUploading, setIsUploading] = useState(false)
  const [uploadProgress, setUploadProgress] = useState(0)
  const fileInputRef = useRef<HTMLInputElement>(null)
  const notifications = useNotifications()

  const validateFile = (file: File): string | null => {
    if (!ALLOWED_FILE_TYPES.includes(file.type)) {
      return 'Please select a valid image file (JPEG, PNG, or WebP)'
    }
    
    if (file.size > MAX_FILE_SIZE) {
      return 'File size must be less than 5MB'
    }
    
    return null
  }

  const handleFileSelect = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
    const file = event.target.files?.[0]
    if (!file) return

    const validationError = validateFile(file)
    if (validationError) {
      notifications.formError(validationError)
      return
    }

    setSelectedFile(file)
    
    // Create preview URL
    const url = URL.createObjectURL(file)
    setPreviewUrl(url)
  }, [notifications])

  const handleUpload = async () => {
    if (!selectedFile) return

    setIsUploading(true)
    setUploadProgress(0)

    try {
      const supabase = createClient()
      
      // Create unique filename
      const fileExt = selectedFile.name.split('.').pop()
      const fileName = `${userId}_${Date.now()}.${fileExt}`
      const filePath = `avatars/${fileName}`

      // Upload file to Supabase Storage
      const { data: uploadData, error: uploadError } = await supabase.storage
        .from('avatars')
        .upload(filePath, selectedFile, {
          cacheControl: '3600',
          upsert: false
        })

      if (uploadError) {
        console.error('Upload error:', uploadError)
        notifications.formError(`Upload failed: ${uploadError.message}`)
        return
      }

      setUploadProgress(50)

      // Get public URL
      const { data: urlData } = supabase.storage
        .from('avatars')
        .getPublicUrl(filePath)

      if (!urlData.publicUrl) {
        notifications.formError('Failed to get avatar URL')
        return
      }

      setUploadProgress(75)

      // Update profile in database
      const { error: updateError } = await supabase
        .from('profiles')
        .update({ avatar_url: urlData.publicUrl })
        .eq('user_id', userId)
        .select()

      if (updateError) {
        console.error('Profile update error:', updateError)
        notifications.formError(`Failed to update profile: ${updateError.message}`)
        return
      }

      setUploadProgress(100)

      // Success!
      notifications.formSuccess('Avatar updated successfully!')
      onAvatarUpdate?.(urlData.publicUrl)
      
      // Clean up and close
      setTimeout(() => {
        setIsOpen(false)
        setSelectedFile(null)
        setPreviewUrl(null)
        setUploadProgress(0)
      }, 1000)

    } catch (error) {
      console.error('Avatar upload error:', error)
      notifications.formError('An unexpected error occurred during upload')
    } finally {
      setIsUploading(false)
    }
  }

  const clearSelection = () => {
    setSelectedFile(null)
    if (previewUrl) {
      URL.revokeObjectURL(previewUrl)
      setPreviewUrl(null)
    }
    if (fileInputRef.current) {
      fileInputRef.current.value = ''
    }
  }

  const triggerFileSelect = () => {
    fileInputRef.current?.click()
  }

  return (
    <Dialog open={isOpen} onOpenChange={setIsOpen}>
      <DialogTrigger asChild>
        <Button variant="outline" className={cn("w-full sm:w-auto", className)}>
          <Camera className="mr-2 h-4 w-4" />
          Edit Avatar
        </Button>
      </DialogTrigger>
      
      <DialogContent className="sm:max-w-md">
        <DialogHeader>
          <DialogTitle>Update Avatar</DialogTitle>
        </DialogHeader>
        
        <div className="space-y-6">
          {/* Current Avatar Display */}
          <div className="flex flex-col items-center space-y-4">
            <Avatar className="h-24 w-24">
              <AvatarImage 
                src={previewUrl || currentAvatarUrl || undefined} 
                alt={userDisplayName} 
              />
              <AvatarFallback className="text-lg font-semibold bg-gradient-to-br from-blue-500 to-purple-600 text-white">
                {getInitials(userDisplayName)}
              </AvatarFallback>
            </Avatar>
            
            {selectedFile && (
              <div className="text-center">
                <p className="text-sm text-muted-foreground">
                  {selectedFile.name} ({(selectedFile.size / 1024 / 1024).toFixed(2)} MB)
                </p>
              </div>
            )}
          </div>

          {/* File Input */}
          <input
            ref={fileInputRef}
            type="file"
            accept={ALLOWED_FILE_TYPES.join(',')}
            onChange={handleFileSelect}
            className="hidden"
          />

          {/* Upload Progress */}
          {isUploading && (
            <div className="space-y-2">
              <div className="flex items-center justify-between text-sm">
                <span>Uploading...</span>
                <span>{uploadProgress}%</span>
              </div>
              <div className="w-full bg-gray-200 rounded-full h-2">
                <div 
                  className="bg-blue-600 h-2 rounded-full transition-all duration-300"
                  style={{ width: `${uploadProgress}%` }}
                />
              </div>
            </div>
          )}

          {/* Action Buttons */}
          <div className="flex flex-col sm:flex-row gap-3">
            {!selectedFile ? (
              <Button 
                onClick={triggerFileSelect}
                className="flex-1"
                disabled={isUploading}
              >
                <Upload className="mr-2 h-4 w-4" />
                Choose Image
              </Button>
            ) : (
              <>
                <Button 
                  onClick={handleUpload}
                  disabled={isUploading}
                  className="flex-1"
                >
                  {isUploading ? (
                    <>
                      <Loader2 className="mr-2 h-4 w-4 animate-spin" />
                      Uploading...
                    </>
                  ) : uploadProgress === 100 ? (
                    <>
                      <Check className="mr-2 h-4 w-4" />
                      Complete!
                    </>
                  ) : (
                    <>
                      <Upload className="mr-2 h-4 w-4" />
                      Upload
                    </>
                  )}
                </Button>
                
                <Button 
                  variant="outline" 
                  onClick={clearSelection}
                  disabled={isUploading}
                  className="sm:w-auto"
                >
                  <X className="mr-2 h-4 w-4" />
                  Clear
                </Button>
              </>
            )}
          </div>

          {/* Help Text */}
          <div className="text-center">
            <p className="text-xs text-muted-foreground">
              Supported formats: JPEG, PNG, WebP â€¢ Max size: 5MB
            </p>
          </div>
        </div>
      </DialogContent>
    </Dialog>
  )
} 