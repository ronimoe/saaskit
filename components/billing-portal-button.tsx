'use client';

import { useState } from 'react';
import { useRouter } from 'next/navigation';
import { createClient } from '@/utils/supabase/client';
import { Button } from '@/components/ui/button';
import { CreditCard, ExternalLink, Loader2 } from 'lucide-react';
import { toast } from 'sonner';

interface BillingPortalButtonProps {
  variant?: 'default' | 'outline' | 'secondary' | 'ghost' | 'link' | 'destructive';
  size?: 'default' | 'sm' | 'lg' | 'icon';
  className?: string;
  children?: React.ReactNode;
  showIcon?: boolean;
}

export function BillingPortalButton({
  variant = 'default',
  size = 'default',
  className,
  children,
  showIcon = true,
}: BillingPortalButtonProps) {
  const [isLoading, setIsLoading] = useState(false);
  const router = useRouter();
  const supabase = createClient();

  const handlePortalAccess = async () => {
    try {
      setIsLoading(true);

      // Check if user is authenticated
      const { data: { user }, error: authError } = await supabase.auth.getUser();
      
      if (authError || !user) {
        toast.error('Please sign in to manage your billing');
        router.push('/login');
        return;
      }

      console.log('[PORTAL BUTTON] Initiating portal session for user:', user.id);

      // Call the portal API
      const response = await fetch('/api/stripe/portal', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          userId: user.id,
        }),
      });

      const data = await response.json();

      if (!response.ok) {
        if (data.type === 'configuration_error') {
          throw new Error(`${data.error}. ${data.details || ''}`);
        }
        throw new Error(data.error || 'Failed to access billing portal');
      }

      if (!data.success || !data.url) {
        throw new Error('Invalid response from billing portal');
      }

      console.log('[PORTAL BUTTON] Redirecting to portal:', data.sessionId);

      // Redirect to Stripe Customer Portal
      window.location.href = data.url;

    } catch (error) {
      console.error('[PORTAL BUTTON] Error accessing portal:', error);
      
      const errorMessage = error instanceof Error ? error.message : 'Failed to access billing portal';
      
      if (errorMessage.includes('No billing account found')) {
        toast.error('No billing account found. Please create a subscription first.');
        router.push('/pricing');
      } else if (errorMessage.includes('Stripe Customer Portal is not configured')) {
        // Handle configuration error specifically
        toast.error('Stripe Customer Portal is not configured. Redirecting to Stripe dashboard instead.');
        
        // Optionally, if you're in development mode, show more details
        if (process.env.NODE_ENV === 'development') {
          console.warn('Developer Note: Configure the Stripe Customer Portal at https://dashboard.stripe.com/test/settings/billing/portal');
        }
        
        // Get user again for the fallback
        const { data: { user: currentUser } } = await supabase.auth.getUser();
        if (!currentUser) return;
        
        // Try to get the Stripe customer ID and redirect to Stripe dashboard as fallback
        try {
          const customerResponse = await fetch('/api/stripe/get-customer-id', {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ userId: currentUser.id }),
          });
          
          const customerData = await customerResponse.json();
          
          if (customerResponse.ok && customerData.stripeCustomerId) {
            // Redirect to Stripe dashboard for this customer
            window.open(`https://dashboard.stripe.com/test/customers/${customerData.stripeCustomerId}`, '_blank');
          } else {
            // If we can't get customer ID, just redirect to Stripe dashboard
            window.open('https://dashboard.stripe.com/test/customers', '_blank');
          }
        } catch (fallbackError) {
          console.error('Error with fallback redirect:', fallbackError);
          window.open('https://dashboard.stripe.com/test', '_blank');
        }
      } else {
        toast.error(errorMessage);
      }
    } finally {
      setIsLoading(false);
    }
  };

  return (
    <Button
      onClick={handlePortalAccess}
      disabled={isLoading}
      variant={variant}
      size={size}
      className={className}
    >
      {isLoading ? (
        <Loader2 className="h-4 w-4 animate-spin" />
      ) : (
        showIcon && <CreditCard className="h-4 w-4" />
      )}
      <span className={showIcon ? 'ml-2' : ''}>
        {children || 'Manage Billing'}
      </span>
      {!isLoading && showIcon && <ExternalLink className="h-4 w-4 ml-2" />}
    </Button>
  );
} 