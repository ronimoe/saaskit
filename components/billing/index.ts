export { PlanComparison } from './plan-comparison'
export { PaymentHistory } from './payment-history'
export { BillingAddressForm } from './billing-address-form'
export { SubscriptionManagement } from './subscription-management' 