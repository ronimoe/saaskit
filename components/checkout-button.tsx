'use client';

import { useState } from 'react';
import { useRouter } from 'next/navigation';
import { createClient } from '@/utils/supabase/client';
import { getStripe } from '@/lib/stripe-client';
import { Button } from '@/components/ui/button';
import { Loader2, UserPlus } from 'lucide-react';

interface CheckoutButtonProps {
  priceId: string;
  planName: string;
  isPopular?: boolean;
  className?: string;
  children?: React.ReactNode;
  enableGuestCheckout?: boolean;
}

export default function CheckoutButton({ 
  priceId, 
  planName, 
  isPopular = false, 
  className = '',
  children,
  enableGuestCheckout = true
}: CheckoutButtonProps) {
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState<string | null>(null);
  const [checkingAuth, setCheckingAuth] = useState(false);
  const router = useRouter();

  const handleCheckout = async () => {
    try {
      setIsLoading(true);
      setError(null);
      setCheckingAuth(true);

      const supabase = createClient();
      const { data: { user } } = await supabase.auth.getUser();
      
      setCheckingAuth(false);

      let checkoutData;

      if (user) {
        // Authenticated user checkout
        checkoutData = {
          priceId,
          userId: user.id,
          userEmail: user.email,
          fullName: user.user_metadata?.full_name || user.user_metadata?.name || undefined,
          isGuest: false
        };

        console.log('[CHECKOUT BUTTON] Starting authenticated checkout:', {
          priceId,
          userId: user.id,
          hasEmail: !!user.email,
          hasFullName: !!checkoutData.fullName
        });
      } else {
        // Guest checkout (no authentication required)
        if (!enableGuestCheckout) {
          router.push('/login?redirect=/pricing');
          return;
        }

        checkoutData = {
          priceId,
          isGuest: true,
          planName
        };

        console.log('[CHECKOUT BUTTON] Starting guest checkout:', {
          priceId,
          planName,
          isGuest: true
        });
      }

      // Create checkout session
      const response = await fetch('/api/stripe/checkout', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(checkoutData),
      });

      if (!response.ok) {
        const errorData = await response.json();
        throw new Error(errorData.error || 'Failed to create checkout session');
      }

      const { sessionId, url, customerId } = await response.json();

      console.log('[CHECKOUT BUTTON] Checkout session created:', {
        sessionId,
        customerId: customerId || 'guest',
        hasUrl: !!url,
        isGuest: !user
      });

      if (url) {
        // Redirect directly to Stripe Checkout
        window.location.href = url;
      } else if (sessionId) {
        // Fallback: use Stripe.js to redirect
        const stripe = await getStripe();
        if (!stripe) {
          throw new Error('Failed to load Stripe');
        }

        const { error: stripeError } = await stripe.redirectToCheckout({ sessionId });
        
        if (stripeError) {
          throw new Error(stripeError.message);
        }
      } else {
        throw new Error('No checkout URL or session ID received');
      }

    } catch (err) {
      console.error('Checkout error:', err);
      setError(err instanceof Error ? err.message : 'Failed to start checkout');
    } finally {
      setIsLoading(false);
      setCheckingAuth(false);
    }
  };

  const defaultClassName = `w-full ${
    isPopular 
      ? 'bg-purple-600 hover:bg-purple-700' 
      : 'bg-gray-900 hover:bg-gray-800'
  } text-white`;

  // Show different text and icons based on loading state
  const getButtonContent = () => {
    if (isLoading) {
      if (checkingAuth) {
        return (
          <>
            <Loader2 className="w-4 h-4 animate-spin mr-2" />
            Checking account...
          </>
        );
      }
      return (
        <>
          <Loader2 className="w-4 h-4 animate-spin mr-2" />
          Starting checkout...
        </>
      );
    }

    if (children) {
      return children;
    }

    // Default button content with guest checkout indication
    if (enableGuestCheckout) {
      return (
        <div className="flex items-center justify-center">
          <UserPlus className="w-4 h-4 mr-2" />
          Start Free Trial
        </div>
      );
    }

    return `Subscribe to ${planName}`;
  };

  return (
    <div className="space-y-2">
      <Button 
        onClick={handleCheckout}
        disabled={isLoading}
        className={className || defaultClassName}
        size="lg"
      >
        {getButtonContent()}
      </Button>
      
      {enableGuestCheckout && !isLoading && (
        <p className="text-xs text-gray-500 text-center">
          No account required â€¢ Create account after payment
        </p>
      )}
      
      {error && (
        <p className="text-sm text-red-600 text-center">
          {error}
        </p>
      )}
    </div>
  );
} 