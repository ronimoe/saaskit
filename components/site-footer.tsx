import { Button } from "@/components/ui/button"
import { Separator } from "@/components/ui/separator"
import { Badge } from "@/components/ui/badge"
import { Mail, Github, Globe, Code2, Zap, Heart } from "lucide-react"
import Link from "next/link"

export function SiteFooter() {
  return (
    <footer className="mt-20 bg-gradient-to-b from-slate-50 to-white dark:from-slate-900 dark:to-slate-950">
      <div className="container mx-auto px-4 py-16">
        {/* Main Footer Content */}
        <div className="grid grid-cols-1 md:grid-cols-4 gap-8 mb-12">
          
          {/* Brand Section */}
          <div className="md:col-span-2">
            <div className="flex items-center gap-2 mb-4">
              <div className="w-8 h-8 rounded-lg bg-gradient-to-r from-blue-600 to-purple-600 flex items-center justify-center">
                <Code2 className="w-4 h-4 text-white" />
              </div>
              <h3 className="text-xl font-bold bg-gradient-to-r from-slate-900 to-slate-600 dark:from-white dark:to-slate-300 bg-clip-text text-transparent">
                SaaS Kit
              </h3>
            </div>
            <p className="text-slate-600 dark:text-slate-400 mb-6 max-w-md leading-relaxed">
              The complete Next.js SaaS starter kit with authentication, payments, database, and beautiful UI components. 
              Ship your SaaS faster with production-ready code.
            </p>
            <div className="flex flex-wrap gap-2">
              <Badge variant="secondary" className="bg-blue-100 text-blue-800 dark:bg-blue-900 dark:text-blue-200">
                <Zap className="w-3 h-3 mr-1" />
                Next.js 15
              </Badge>
              <Badge variant="secondary" className="bg-purple-100 text-purple-800 dark:bg-purple-900 dark:text-purple-200">
                TypeScript
              </Badge>
              <Badge variant="secondary" className="bg-green-100 text-green-800 dark:bg-green-900 dark:text-green-200">
                Supabase
              </Badge>
            </div>
          </div>

          {/* Quick Links */}
          <div>
            <h4 className="font-semibold text-slate-900 dark:text-slate-100 mb-4">Quick Links</h4>
            <div className="space-y-3">
              <Link href="/pricing" className="block text-slate-600 dark:text-slate-400 hover:text-slate-900 dark:hover:text-slate-100 transition-colors">
                Pricing
              </Link>
              <Link href="/login" className="block text-slate-600 dark:text-slate-400 hover:text-slate-900 dark:hover:text-slate-100 transition-colors">
                Sign In
              </Link>
              <Link href="/signup" className="block text-slate-600 dark:text-slate-400 hover:text-slate-900 dark:hover:text-slate-100 transition-colors">
                Get Started
              </Link>
            </div>
          </div>

          {/* Resources & Contact */}
          <div>
            <h4 className="font-semibold text-slate-900 dark:text-slate-100 mb-4">Resources</h4>
            <div className="space-y-3">
              <a 
                href="https://github.com/ronimoe/saaskit" 
                target="_blank" 
                rel="noopener noreferrer"
                className="flex items-center gap-2 text-slate-600 dark:text-slate-400 hover:text-slate-900 dark:hover:text-slate-100 transition-colors group"
              >
                <Github className="w-4 h-4 group-hover:scale-110 transition-transform" />
                GitHub Repository
              </a>
              <div className="flex items-center gap-2 text-slate-600 dark:text-slate-400">
                <Mail className="w-4 h-4" />
                hello@saaskit.com
              </div>
              <div className="flex items-center gap-2 text-slate-600 dark:text-slate-400">
                <Globe className="w-4 h-4" />
                www.saaskit.com
              </div>
            </div>
          </div>
        </div>

        <Separator className="mb-8" />

        {/* Bottom Section */}
        <div className="flex flex-col md:flex-row justify-between items-center gap-4">
          <div className="flex items-center gap-2 text-sm text-slate-500 dark:text-slate-400">
            <span>Â© 2025 SaaS Kit. Built with</span>
            <Heart className="w-4 h-4 text-red-500" />
            <span>for the love of code</span>
          </div>
          
          {/* GitHub Stars Button */}
          <Button 
            variant="outline" 
            size="sm" 
            className="border-slate-200 dark:border-slate-700 hover:bg-slate-50 dark:hover:bg-slate-800" 
            asChild
          >
            <a 
              href="https://github.com/ronimoe/saaskit" 
              target="_blank" 
              rel="noopener noreferrer"
              className="flex items-center gap-2"
            >
              <Github className="w-4 h-4" />
              Star on GitHub
            </a>
          </Button>
        </div>
      </div>
    </footer>
  )
} 