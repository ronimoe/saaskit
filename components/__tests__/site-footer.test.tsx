import { render } from '@testing-library/react';
import { SiteFooter } from '@/components/site-footer';

describe('SiteFooter', () => {
  it('should match snapshot', () => {
    const { container } = render(<SiteFooter />);
    expect(container).toMatchSnapshot();
  });
}); 