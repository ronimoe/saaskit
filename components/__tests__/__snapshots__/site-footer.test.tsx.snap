// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteFooter should match snapshot 1`] = `
<div>
  <footer
    class="mt-20 bg-gradient-to-b from-slate-50 to-white dark:from-slate-900 dark:to-slate-950"
  >
    <div
      class="container mx-auto px-4 py-16"
    >
      <div
        class="grid grid-cols-1 md:grid-cols-4 gap-8 mb-12"
      >
        <div
          class="md:col-span-2"
        >
          <div
            class="flex items-center gap-2 mb-4"
          >
            <div
              class="w-8 h-8 rounded-lg bg-gradient-to-r from-blue-600 to-purple-600 flex items-center justify-center"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-code-xml w-4 h-4 text-white"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m18 16 4-4-4-4"
                />
                <path
                  d="m6 8-4 4 4 4"
                />
                <path
                  d="m14.5 4-5 16"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold bg-gradient-to-r from-slate-900 to-slate-600 dark:from-white dark:to-slate-300 bg-clip-text text-transparent"
            >
              SaaS Kit
            </h3>
          </div>
          <p
            class="text-slate-600 dark:text-slate-400 mb-6 max-w-md leading-relaxed"
          >
            The complete Next.js SaaS starter kit with authentication, payments, database, and beautiful UI components. Ship your SaaS faster with production-ready code.
          </p>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent [a&]:hover:bg-secondary/90 bg-blue-100 text-blue-800 dark:bg-blue-900 dark:text-blue-200"
              data-slot="badge"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-zap w-3 h-3 mr-1"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
                />
              </svg>
              Next.js 15
            </span>
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent [a&]:hover:bg-secondary/90 bg-purple-100 text-purple-800 dark:bg-purple-900 dark:text-purple-200"
              data-slot="badge"
            >
              TypeScript
            </span>
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent [a&]:hover:bg-secondary/90 bg-green-100 text-green-800 dark:bg-green-900 dark:text-green-200"
              data-slot="badge"
            >
              Supabase
            </span>
          </div>
        </div>
        <div>
          <h4
            class="font-semibold text-slate-900 dark:text-slate-100 mb-4"
          >
            Quick Links
          </h4>
          <div
            class="space-y-3"
          >
            <a
              class="block text-slate-600 dark:text-slate-400 hover:text-slate-900 dark:hover:text-slate-100 transition-colors"
              href="/pricing"
            >
              Pricing
            </a>
            <a
              class="block text-slate-600 dark:text-slate-400 hover:text-slate-900 dark:hover:text-slate-100 transition-colors"
              href="/login"
            >
              Sign In
            </a>
            <a
              class="block text-slate-600 dark:text-slate-400 hover:text-slate-900 dark:hover:text-slate-100 transition-colors"
              href="/signup"
            >
              Get Started
            </a>
          </div>
        </div>
        <div>
          <h4
            class="font-semibold text-slate-900 dark:text-slate-100 mb-4"
          >
            Resources
          </h4>
          <div
            class="space-y-3"
          >
            <a
              class="flex items-center gap-2 text-slate-600 dark:text-slate-400 hover:text-slate-900 dark:hover:text-slate-100 transition-colors group"
              href="https://github.com/ronimoe/saaskit"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-github w-4 h-4 group-hover:scale-110 transition-transform"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
                />
                <path
                  d="M9 18c-4.51 2-5-2-7-2"
                />
              </svg>
              GitHub Repository
            </a>
            <div
              class="flex items-center gap-2 text-slate-600 dark:text-slate-400"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-mail w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m22 7-8.991 5.727a2 2 0 0 1-2.009 0L2 7"
                />
                <rect
                  height="16"
                  rx="2"
                  width="20"
                  x="2"
                  y="4"
                />
              </svg>
              hello@saaskit.com
            </div>
            <div
              class="flex items-center gap-2 text-slate-600 dark:text-slate-400"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-globe w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <path
                  d="M12 2a14.5 14.5 0 0 0 0 20 14.5 14.5 0 0 0 0-20"
                />
                <path
                  d="M2 12h20"
                />
              </svg>
              www.saaskit.com
            </div>
          </div>
        </div>
      </div>
      <div
        class="bg-border shrink-0 data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px mb-8"
        data-orientation="horizontal"
        data-slot="separator"
        role="none"
      />
      <div
        class="flex flex-col md:flex-row justify-between items-center gap-4"
      >
        <div
          class="flex items-center gap-2 text-sm text-slate-500 dark:text-slate-400"
        >
          <span>
            Â© 2025 SaaS Kit. Built with
          </span>
          <svg
            aria-hidden="true"
            class="lucide lucide-heart w-4 h-4 text-red-500"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
            />
          </svg>
          <span>
            for the love of code
          </span>
        </div>
        <a
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:text-accent-foreground dark:bg-input/30 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5 border-slate-200 dark:border-slate-700 hover:bg-slate-50 dark:hover:bg-slate-800 flex items-center gap-2"
          data-slot="button"
          href="https://github.com/ronimoe/saaskit"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-github w-4 h-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
            />
            <path
              d="M9 18c-4.51 2-5-2-7-2"
            />
          </svg>
          Star on GitHub
        </a>
      </div>
    </div>
  </footer>
</div>
`;
